### DISCUSSION TITLE
can't upload a file with folder

### DISCUSSION THREAD

**Discussion by itsskyballs** [community/community#77390] [Repositories] [answered]
### Select Topic Area

Bug

### Body

I can't upload a folder(contains of my source code files) to my repository source code and some people may confuse it's structure of my repository
![Screenshot_20231130_114216](https://github.com/community/community/assets/111575685/74960f93-32f0-47bc-a9ec-253c0e08a952)

**Comment by ghost** [comment]
you can't directly upload a folder through the GitHub web interface. You need to use git commands on your local machine(pc or laptop  ) to add, commit and push the folder to your repository. To do so download git and initialize you existing folder and commit to the github repo. I'm a git user

**Reply by itsskyballs** [reply]
so will I structure the source code for myself huh?

**Comment by ghost** [comment]
Or try adding files separately into a repostory's folder as I guess you are in a mobile device

**Comment by ghost** [comment]
If it was useful mark mine as answer it would help me alot

**Comment by Thiwanka570** [comment, marked as answer]
you can use git gui

**Reply by itsskyballs** [reply]
yea but I don't have pc,I tried on Termux and it says no such file or directory/access is denied 
because my phone isn't rooted

**Reply by itsskyballs** [reply]
already installed git

**Comment by ghost** [comment]
View this video this might help you

https://youtu.be/eGaImwD8fPQ?si=if1JTW9uCSfuH4yr

**Reply by itsskyballs** [reply]
I'm not gonna watch it because my YouTube premium is rip,I gues I will try it on incognito
